docs/guides/migration-of-compute.md

```diff
@@ -0,0 +1,217 @@
+
+## Example configuration
+
+In this guide, we will assume the following configuration as an example:
+
+```hcl
+resource "exoscale_compute" "my_instance" {
+  disk_size = 10
+  display_name = "my-instance"
+  security_group_ids = ["493ac3c0-c8ba-415a-a038-6578194a6d36"]
+  size = "Tiny"
+  template_id = "ee73810b-0245-43e0-8b15-0632473d56ba"
+  zone = "ch-gva-2"
+}
+
+resource "exoscale_network" "my_network" {
+  name = "privnet"
+  display_text = "Private Network"
+  zone = "ch-gva-2"
+}
+
+resource "exoscale_nic" "my_nic" {
+  compute_id = exoscale_compute.my_instance.id
+  network_id = exoscale_network.my_network.id
+}
+```
+
+As you can see, we have a single `exoscale_compute` (instance) attached to the `exoscale_network` (private network) using `exoscale_nic` resource.
+
+The Terraform state contains them as well:
+
+```bash
+$ terraform state list
+exoscale_compute.my_instance
+exoscale_network.my_network
+exoscale_nic.my_nic
+```
+
+We want to migrate `exoscale_compute` and `exoscale_nic` to the `exoscale_compute_instance` resource and `exoscale_network` to the `exoscale_private_network` resource.
+
+
+## Migration plan
+
+To achieve this migration, we have to remove all 3 deprecated resources from the state and then import them as `exoscale_compute_instance` and `exoscale_private_network`.
+
+## Applying the migration plan
+
+### Removing the security group from the state 
+
+This step is pretty straightforward: we will have to issue a `terraform state rm` command for each resources we have to remove from the state. In our example, we have to remove `exoscale_compute.my_instance`, `exoscale_network.my_network` and `exoscale_nic.my_nic`:
+
+```bash
+$ terraform state rm exoscale_compute.my_instance
+Removed exoscale_compute.my_instance
+Successfully removed 1 resource instance(s).
+
+q$ terraform state rm exoscale_network.my_network
+Removed exoscale_network.my_network
+Successfully removed 1 resource instance(s).
+
+$ terraform state rm exoscale_nic.my_nic
+Removed exoscale_nic.my_nic
+Successfully removed 1 resource instance(s).
+```
+
+Now, these resources are removed from the state.
+
+### Update infrastructure definition
+
+Replace the `exoscale_network` block by the new `exoscale_private_network` resource:
+
+```hcl
+resource "exoscale_private_network" "my_network" {
+  name = "privnet"
+  description = "Private Network"
+  zone = "ch-gva-2"
+}
+```
+
+In this example we are using unmanaged private network, for managed network you can copy values from `exoscale_network` definition with the following changes:
+- `display_text` must be renamed to `description`;
+- `network_offering` and `tags` are deprecated and should be removed.
```

Comment:
`labels` exist in public-api to replace `tags`, we should implement them for the `exoscale_private_network` resource